### PR TITLE
Add an initializer

### PR DESCRIPTION
**Pull request by AW-UC**
This adds an extensible initializer and a basic test for it

### CODE REVIEW

File: tests/unit/initializers/ember-leaflet-test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      container = application.__container__;
+      application.deferReadiness();
+    });
+  }
+});
+
+test('it works', function(assert) {
+  initialize(container, application);
+
+  assert.ok(typeof L.Icon.Default.imagePath !== 'undefined', "'L.Icon.Default.imagePath' is not set");
```
Comment (miguelcobain) [inline tests/unit/initializers/ember-leaflet-test.js]: Shouldn't the message be "'L.Icon.Default.imagePath' **is** set"?

Comment (AW-UC) [inline tests/unit/initializers/ember-leaflet-test.js]: No, that is the message it shows, if the test fails. Thus, if it's type is `=== undefined`

**Comment by miguelcobain** [comment]
Superb, @AW-UC!

**Comment by gabesmed** [comment]
why are we setting imagePath in Ember Leaflet? shouldn't that be the responsibility of whatever app uses it? I'm super wary of a hard-coded "assets/images" string inside a general-purpose library -- we should be agnostic to where you put your assets.

**Comment by AW-UC** [comment]
@gabesmed they are indeed pros and cons for either way. the alternative to setting this value would be adding an entry in the ReadMe that tells the user to set it himself.
However, I am in favor of hardcoding this, because Ember CLI follows the paradigm of *convention over configuration*. Thus, if you initialize a new Ember app and install `ember-leaflet`, the marker PNGs etc. will reside in `public/assets/images`. This is just the recommended standard path for that stuff (see: http://www.ember-cli.com/asset-compilation/). So the majority of users will be fine with that. If you want to change it, you can easily overide that setting with your own initializer. We could add an entry on how to do that in the ReadMe.

**Comment by miguelcobain** [comment]
I think that the main argument in favor of this is that an ember-cli addon does more than exposing a codebase to the consuming app. It also takes care of including assets.
However I do agree that we should expose a configuration to the consuming app and an entry in the README.

Example of such a configuration: 
```javascript
//your-app/Brocfile.js

var EmberApp = require('ember-cli/lib/broccoli/ember-app');

var app = new EmberApp({
  'ember-leaflet': {
    assetsPath: 'assets/images'
  }
});

module.exports = app.toTree();
```

I think this is the correct ember-cli way of doing this.
Still, any consuming app can override this initializer, I think.

**Comment by AW-UC** [comment]
The option to define the assets path in Brocfile.js would be a nice addition. It should also default to some value (like `assets/images`).

**Pull request merged**